fix(upgrade): don't create BE if up to date

ELDiablO59152 opened the pull request:
### Fixed
- [UPGRADE-OS] Wrongly create new BE when already up to date
- [UPGRADE-OS] Incorrect date format
- [UPGRADE-OS] Correctly Download and use the upgrade when upgrading only one or more jails and base is up-to-date

ELDiablO59152 committed:
fix(upgrade): Only create a new BE when necessary

frikilax commented on usr/local/share/vulture-utils/upgrade-os.sh:
```diff
@@ -74,11 +74,8 @@ update_system() {
         fi
         if [ $snapshot_system -gt 0 ]; then
-            /sbin/bectl create "${_snap_name}" || finalize 1 "Could not create a new Boot Environment!"
             clean_old_BEs "$keep_previous_snap"
```
```suggestion
```
This should not be done here unconditionaly

ELDiablO59152 replied:
Done

frikilax commented on usr/local/share/vulture-utils/upgrade-os.sh:
```diff
@@ -164,7 +149,7 @@ finalize() {
     has_pending_BE
```
```suggestion
    if has_pending_BE; then
        clean_old_BEs "$keep_previous_snap"
    fi
```

ELDiablO59152 replied:
Suggestion applied

frikilax committed:
clean(upgrade-os): Use the same function when upgrading jail's OS

frikilax committed:
clean(common): Remove obsolete functions

Those functions are defined in the corresponding upgrade-os.sh file, and only used there

frikilax committed:
fix(upgrade-os): Correctly Download and use the upgrade when upgrading only one or more jails and base is up-to-date
